front_end/webserver.py

```diff
@@ -300,72 +339,115 @@ def get(self, course, assignment, problem):
-            problem_details["expected_output"] = ""
-            problem_details["data_urls_info"] = []
-
-            if problem_basics["title"] == "" or problem_details["instructions"] == "" or problem_details["answer_code"] == "":
+            new_problem = True
+            if content.check_problem_exists(problem):
+                new_problem = False
+                problem_details = content.get_problem_details(course, assignment, problem)
+
+            problem_basics = content.get_problem_basics(course, assignment, problem)
+            
+            if new_problem:
+                problem_basics["title"] = self.get_body_argument("title").strip() #required
+                problem_basics["visible"] = self.get_body_argument("is_visible") == "Yes" #required
+                problem_details = {}
+                problem_details["instructions"] = self.get_body_argument("instructions").strip().replace("\r", "") #required
+                problem_details["back_end"] = self.get_body_argument("back_end")
+                problem_details["output_type"] = self.get_body_argument("output_type")
+                problem_details["answer_code"] = self.get_body_argument("answer_code").strip().replace("\r", "") #required
+                problem_details["answer_description"] = self.get_body_argument("answer_description").strip().replace("\r", "")
+                problem_details["test_code"] = self.get_body_argument("test_code").strip().replace("\r", "")
+                problem_details["credit"] = self.get_body_argument("credit").strip().replace("\r", "")
+                problem_details["data_url"] = self.get_body_argument("data_url").strip().replace("\r", "")
+                problem_details["show_expected"] = self.get_body_argument("show_expected") == "Yes"
+                problem_details["show_test_code"] = self.get_body_argument("show_test_code") == "Yes"
+                problem_details["show_answer"] = self.get_body_argument("show_answer") == "Yes"
+                problem_details["expected_output"] = ""
+
+            if self.get_body_argument("title").strip() == "" or self.get_body_argument("instructions").strip().replace("\r", "") == "" or self.get_body_argument("answer_code").strip().replace("\r", "") == "":
                 result = "Error: One of the required fields is missing."
             else:
-                if has_duplicate_title(get_problems(course, assignment), problem, problem_basics["title"]):
+                if content.has_duplicate_title(content.get_problems(course, assignment), problem, self.get_body_argument("title").strip()):
                     result = "Error: A problem with that title already exists in this assignment."
                 else:
-                    for data_url in set(problem_details["data_urls"].split("\n")):
-                        data_url = data_url.strip()
-                        if data_url != "":
-                            contents, content_type, extension = download_file(data_url)
```

Comment:
We need to keep this logic in the new version of the code. It just needs to be simplified a bit (not use a loop) because there will only be a max of one data_url.